lib/api/src/index.tsx

```diff
@@ -442,7 +442,11 @@ export function useAddonState<S>(addonId: string, defaultState?: S) {
   return useSharedState<S>(addonId, defaultState);
 }
 
-export function useArgs(): [Args, (newArgs: Args) => void, (argNames?: string[]) => void] {
+export function useArgs<Args extends Record<string, any> = { [key: string]: any }>(): [
```

Comment:
Ow I now just read @shilman 's comment and looked at the changed code.

I'm afraid we can't do this. @Gabriel5934 
The reason is that it would make the code less sound.. Though it's nice to get auto-completion, it's really dangerous to tell yourself you'll receive args of type X, but then at runtime you get a completely different object.

Now typescript tells you everything is fine, but in actuality it will likely throw at runtime.
And in this case when the code throws... it's a fatal problem for the storybook manager code.

We really appreciate your time, energy and effort that you put into this PR, but unfortunately we can't merge it :(
I hope you understand our reasoning. ❤️